test-suite/Math/NumberTheory/Moduli/SqrtTests.hs

```diff
@@ -23,68 +23,75 @@ import Test.Tasty.HUnit
 import Data.List (nub)
 import Data.Maybe (fromJust)
 
-import Math.NumberTheory.Moduli hiding (invertMod)
+import Math.NumberTheory.Moduli hiding (invertMod, unPrime)
 import Math.NumberTheory.UniqueFactorisation (unPrime, isPrime)
 import Math.NumberTheory.TestUtils
 
+fromJustCustom :: Maybe a -> a
+fromJustCustom Nothing = error "Not a prime"
+fromJustCustom (Just a) = a
+
 unwrapP :: PrimeWrapper Integer -> Integer
 unwrapP (PrimeWrapper p) = unPrime p
 
 unwrapPP :: (PrimeWrapper Integer, Power Int) -> (Integer, Int)
 unwrapPP (p, Power e) = (unwrapP p, e)
 
+-- FIXME: update names. Maybe not - they describe specific behavior inside interfaces.
+
 -- | Check that 'sqrtMod' is defined iff a quadratic residue exists.
 --   Also check that the result is a solution of input modular equation.
 sqrtModPProperty :: AnySign Integer -> PrimeWrapper Integer -> Bool
-sqrtModPProperty (AnySign n) (unwrapP -> p) = case sqrtModP n p of
+sqrtModPProperty (AnySign n) (unwrapP -> p) = case sqrtModMaybe n (fromJustCustom . toFieldCharacteristic $ p) of
```

Comment:
We can avoid partial functions here. Change `unwrapP` to return `Prime Integer` and construct `FieldCharacteristic p 1` directly.

Reply:
`Math.NumberTheory.Primes.Types` is hidden by some reason. Should I expose it in order to use `Prime` in tests?